packages/superset-ui-encodable/src/encoders/ChannelEncoderAxis.ts

```diff
@@ -0,0 +1,57 @@
+import ChannelEncoder from './ChannelEncoder';
+import createFormatterFromFieldTypeAndFormat from '../parsers/format/createFormatterFromFieldTypeAndFormat';
+import { CompleteAxisConfig } from '../fillers/completeAxisConfig';
+import { ChannelDef } from '../types/ChannelDef';
+import { Value, DateTime, isDateTime } from '../types/VegaLite';
+import { CompleteFieldDef } from '../fillers/completeChannelDef';
+import { ChannelInput } from '../types/Channel';
+import { HasToString } from '../types/Base';
+import parseDateTime from '../parsers/parseDateTime';
+
+export default class ChannelEncoderAxis<
+  Def extends ChannelDef<Output>,
+  Output extends Value = Value
+> {
+  readonly channelEncoder: ChannelEncoder<Def, Output>;
+  readonly config: Exclude<CompleteAxisConfig, false>;
+  readonly formatValue: (value: ChannelInput | HasToString) => string;
+
+  constructor(channelEncoder: ChannelEncoder<Def, Output>) {
+    this.channelEncoder = channelEncoder;
+    this.config = channelEncoder.definition.axis as Exclude<CompleteAxisConfig, false>;
+    this.formatValue = createFormatterFromFieldTypeAndFormat(
+      (channelEncoder.definition as CompleteFieldDef<Output>).type,
+      this.config.format || '',
+    );
+  }
+
+  getTitle() {
+    return this.config.title;
+  }
+
+  hasTitle() {
+    return this.config.title !== '';
+  }
+
+  getTickLabels() {
+    const { tickCount, values } = this.config;
+
+    if (typeof values !== 'undefined') {
+      return (values as (
+        | number
```

Comment:
could make a separate `type` (or re-use the `config.values` type?)

Reply:
updated to use util `inferElementTypeFromUnionOfArrayTypes`